packages/regl-worldview/src/commands/Command.js

```diff
@@ -9,45 +9,45 @@
 
 import * as React from "react";
 
-import type { ComponentMouseHandler, MouseEventEnum, RawCommand, Vec4, Color, Ray } from "../types";
-import { getIdFromColor, intToRGB } from "../utils/commandUtils";
+import type {
+  ComponentMouseHandler,
+  GetHitmap,
+  MouseEventEnum,
+  RawCommand,
+  Color,
+  Ray,
+  MouseEventObject,
+} from "../types";
 import { getNodeEnv } from "../utils/common";
 import { type WorldviewContextType } from "../WorldviewContext";
 import WorldviewReactContext from "../WorldviewReactContext";
 
 export const SUPPORTED_MOUSE_EVENTS = ["onClick", "onMouseUp", "onMouseMove", "onMouseDown", "onDoubleClick"];
 
-export type HitmapProp<T> = T & ({ colors: Vec4[] } | { color: Vec4 });
-export type GetObjectFromHitmapId<T> = (objectId: number, hitmapProps: HitmapProps<T>[]) => ?HitmapProps<T>;
-export type GetHitmapProps<T> = (children: T[]) => ?HitmapProp<T>;
 export type MarkerDefault = {
   id?: number,
   points?: Point[],
   color?: Color,
 };
-export type HitmapMarkerDefault = {
-  id?: number,
-  points?: Point[],
-  color?: Vec4,
+
+export type OptionalCommandProps = {
+  [MouseEventEnum]: ComponentMouseHandler,
```

Comment:
this one doesn't look optional?

Reply:
Because of the `[]` these mouse events do become optional